network/nxos/nxos_nxapi.py

```diff
@@ -0,0 +1,251 @@
+    Use the options below to override the default configuration.
+extends_documentation_fragment: eos
+options:
+    state:
+        description:
+            - Set to started or stopped. A state of started will
+              enable NXAPI access, and a state of stopped will
+              disable or shutdown all NXAPI access.
+        requred: false
+        default: started
+    http_port:
+        description:
+            - Port on which the HTTP server will listen.
+        required: false
+        default: 80
+    https_port:
+        description:
+            - Port on which the HTTPS server will listen.
+        required: false
+        default: 443
+    http:
+        description:
+            - Enable/disable HTTP server.
+        required: false
+        default: true
+    https:
+        description:
+            - Enable/disable HTTPS server.
+        required: false
+        default: true
+    sandbox:
+        description:
+            - Enable/disable NXAPI web based UI for entering commands.
+        required: false
+        default: true
+"""
+
+EXAMPLES = """
+    - name: Enable NXAPI access with default configuration
+      nxos_nxapi:
+          provider: {{ provider }}
+
+    - name: Enable NXAPI with no HTTP, HTTPS at port 9443 and sandbox disabled
+      nxos_nxapi:
+          enable_http: false
+          https_port: 9443
+          enable_sandbox: no
+          provider: {{ provider }}
+
+    - name: shutdown NXAPI access
+      nxos_nxapi:
+          state: stopped
+          provider: {{ provider }}
+"""
+
+RETURN = """
+changed:
+    description:
+        - Indicates if commands were sent to the device.
+    returned: always
+    type: boolean
+    sample: false
+
+commands:
+    description:
+        - Set of commands to be executed on remote device. If run in check mode,
+          commands will not be executed.
+    returned: always
+    type: list
+    sample: [
+        'config terminal',
+        'nxapi feature',
+        'nxapi http port 8080'
+    ]
+
+_config:
+    description:
+       - Configuration found on the device prior ro any commands being executed.
+    returned: always
+    type: object
+    sample: {...}
+"""
+
+
+def http_commands(protocol, port, enable, config):
+    port_config = config.get('{0}_port'.format(protocol), None)
+    changed = False
+    commands = []
+    if port_config is None and enable:
+        # enable
+        changed = True
+        commands.append('nxapi {0} port {1}'.format(protocol, port))
+    elif port_config is not None:
+        if not enable:
+            # disable
+            commands.append('no nxapi {0}'.format(protocol))
+            changed = True
+        elif port_config != port:
+            # update port
+            commands.append('nxapi {0} port {1}'.format(protocol, port))
+            changed = True
+    return commands, changed
+
+
+def execute_commands(module, commands):
+    if not module.params.get('check_mode'):
+        module.execute(commands)
```

Comment:
use module.configure instead of module.execute and remove config t from the commands list